druid-shell/src/platform/x11/application.rs

```diff
@@ -74,59 +72,61 @@ struct State {
 
 impl Application {
     pub fn new() -> Result<Application, Error> {
-        let (conn, screen_num) = Connection::connect_with_xlib_display()?;
+        let (conn, screen_num) = XCBConnection::connect(None)?;
```

Comment:
I haven't really investigated this stuff with either library, but is this an equivalent port? The `connect_with_xlib_display` docs say that going through XLib is needed for OpenGL support. I'm not sure if cairo uses OpenGL but if it does - does the new connection also provide that?

Reply:
To my knowledge Cairo does not use OpenGL by default. There is an [experimental feature](https://www.cairographics.org/OpenGL/) for it but otherwise it uses the platform library (X11 on Linux). So it should be fine if there is no OpenGL available, unless we want to use it ourselves.

Reply:
I suspect that we will want OpenGL support eventually. If not for cairo, then for #891-related stuff. But it looks like it will need some work on the x11rb side.

Reply:
I guess you will want to use the `x11` crate for this. (The following "talks `C`", I haven't looked at the Rust bindings of any of this.)

- create you `Display*` via `x11`.
- Get the underlying XCB Connection via `XGetXCBConnection`. This gives you an `xcb_connection_t*`.
- This pointer can be given to `x11rb::XCBConnection::from_raw_xcb_connection(dpy, false)`.
- If you want to handle events through x11rb and libxcb, you should call `XSetEventQueueOwner(dpy, XCBOwnsEventQueue)`. Otherwise, libX11 might randomly eat your events / move them to its own event queue.
- One you drop / `XCloseDisplay` the `Display*`, any further use of the `XCBConnection` might cause demons to fly through your nose, so be careful (by the way: Similar fun can result with cairo, but cairo makes it much easier to accidentally continue using the connection after it was closed / freed.).

Congratulations, you now have a single X11 connection which you can access through all of libX11, libxcb and x11rb.

API-wise, I do not see anything that could be added to x11rb. The two functions I mentioned above live in `libX11.so` and `libX11-xcb.so` and x11rb certainly has no business depending on them.

Reply:
Thanks for the detailed information! I'd prefer not to block this PR on OpenGL support, so I've just dropped in a comment for now.